stream/src/main/scala/org/apache/pekko/stream/scaladsl/Flow.scala

```diff
@@ -1792,6 +1792,30 @@ trait FlowOps[+Out, +Mat] {
    */
   def foldAsync[T](zero: T)(f: (T, Out) => Future[T]): Repr[T] = via(new FoldAsync(zero, f))
 
+  /**
+   * implemented by `Fold`, only emits its result when the upstream completes,
+   * after which it also completes. Applies the given function towards initialized `true`
+   * and next predicate result, yielding the next predicate result.
+   *
+   * If the function `predicate` throws an exception and the supervision decision is
+   * [[pekko.stream.Supervision.Restart]] current value starts at `true` again
+   * the stream will continue.
+   *
+   * Adheres to the [[ActorAttributes.SupervisionStrategy]] attribute.
+   *
+   * '''Emits when''' upstream completes
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * See also [[FlowOps.fold]]
+   */
+  def forall(predicate: Out => Boolean): Repr[Boolean] =
+    via(Fold(true, (acc: Boolean, element) => acc && predicate(element)))
```

Comment:
add `.name("forAll")`, impelment it with `fold` has once drawback, which is we can't do a quick cancel  after one elemement which doesn't match the predicate. maybe we can test it with `collectWhile and fold`?

Reply:
You can see: scala.collection.IterableOnceOps#forall can do a quick break.

Reply:
Yes, I also thought of this, so I used `acc&&predict (element)` (boolean short circuited)，I wonder if `fold` should not be used.

Reply:
empty stream shoud true or false?